File: back/src/main/java/com/back/domain/concert/entity/ConcertDetail.java
package com.back.domain.concert.entity;

import com.back.global.jpa.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
public class ConcertDetail extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long concertDetailId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "concert_id", nullable = false)
    private Concert concert;

    private String urlDetail;
}

File: back/src/main/kotlin/com/back/domain/concert/entity/ConcertDetail.kt
package com.back.domain.concert.entity

import com.back.global.jpa.entity.BaseEntity
import jakarta.persistence.*

@Entity
class ConcertDetail(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "concert_id", nullable = false)
    var concert: Concert,

    var urlDetail: String? = null
) : BaseEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val concertDetailId: Long? = null

    companion object {
        fun create(concert: Concert, urlDetail: String?): ConcertDetail {
            return ConcertDetail(
                concert = concert,
                urlDetail = urlDetail
            )
        }
    }
}
